pkg/webhook/resources/setting/validator.go

```diff
@@ -1142,6 +1169,85 @@ func (v *settingValidator) checkStorageNetworkValueVaild() error {
 	return nil
 }
 
+func (v *settingValidator) checkVlanStatusReady(config *storagenetworkctl.Config) error {
+	vsList, err := v.vsCache.List(labels.Set{
+		utils.KeyClusterNetworkLabel: config.ClusterNetwork,
+	}.AsSelector())
+	if err != nil {
+		return err
+	}
+
+	if len(vsList) > 0 {
+		for _, vs := range vsList {
+			if networkv1.Ready.IsFalse(vs.Status) {
+				return fmt.Errorf("vs %s status is not Ready", vs.Name)
+			}
+		}
+	} else {
+		return fmt.Errorf("vlan status is not present for cluster network %s", config.ClusterNetwork)
+	}
+
+	return nil
+}
+
+func getMatchNodes(vc *networkv1.VlanConfig) ([]string, error) {
+	if vc.Annotations == nil || vc.Annotations[utils.KeyMatchedNodes] == "" {
+		return nil, nil
+	}
+
+	var matchedNodes []string
+	if err := json.Unmarshal([]byte(vc.Annotations[utils.KeyMatchedNodes]), &matchedNodes); err != nil {
+		return nil, err
+	}
+
+	return matchedNodes, nil
+}
+
+func (v *settingValidator) checkVCSpansAllNodes(config *storagenetworkctl.Config) error {
+	var matchedNodes mapset.Set[string]
+	var vlanConfigNodes []string
+
+	vcs, err := v.vcCache.List(labels.Set{
+		utils.KeyClusterNetworkLabel: config.ClusterNetwork,
+	}.AsSelector())
+	if err != nil {
+		return err
+	}
+
+	if len(vcs) > 0 {
+		for _, vc := range vcs {
+			vnodes, err := getMatchNodes(vc)
+			if err != nil {
+				return err
+			}
+			vlanConfigNodes = append(vlanConfigNodes, vnodes...)
+		}
+		matchedNodes = mapset.NewSet[string](vlanConfigNodes...)
+	} else {
+		return fmt.Errorf("vlan config is not present for cluster network %s", config.ClusterNetwork)
+	}
```

Comment:
Ditto.

Reply:
Maybe we can just `matchedNodes.Append(vnodes...)` and get rid of `vlanConfigNodes` entirely.

Reply:
Thanks @starbops taken care.Updated the PR.

Note:the build failure is to due to a latest breakage in harvester master due to the following commit
"https://github.com/harvester/harvester/commit/542f684a7d13b5558bc9125b636e5aab638b8c96"
failure is due to "fmt" package is not imported in that file

Reply:
#6593 has been merged. Let's rebase the PR again. Thank you.

Reply:
@starbops sure done.The pipeline is green now